esphome/components/gl_r01_i2c/sensor.py

```diff
@@ -0,0 +1,55 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import i2c, sensor
+from esphome.const import (
+    CONF_ID,
+    DEVICE_CLASS_DISTANCE,
+    STATE_CLASS_MEASUREMENT,
+    UNIT_MILLIMETER,
+)
+
+CONF_MIN_READ_INTERVAL = "min_read_interval"
+
+gl_r01_i2c_ns = cg.esphome_ns.namespace("gl_r01_i2c")
+GLR01I2CComponent = gl_r01_i2c_ns.class_(
+    "GLR01I2CComponent", i2c.I2CDevice, cg.PollingComponent
+)
+
+
+# Ensure min_read_interval is at least 40ms as stated in datasheet
+def validate_min_read_interval(value):
+    value = cv.positive_time_period_milliseconds(value)
+    if value.total_milliseconds < 40:
+        raise cv.Invalid(f"min_read_interval must be at least 40ms (got {value})")
+    return value
+
+
+CONFIG_SCHEMA = (
+    sensor.sensor_schema(
+        GLR01I2CComponent,
+        unit_of_measurement=UNIT_MILLIMETER,
+        accuracy_decimals=0,
+        device_class=DEVICE_CLASS_DISTANCE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    )
+    .extend(cv.polling_component_schema("1s"))
```

Comment:
Default for sensors is 60s
Users can override this if they want faster updates.

```suggestion
    .extend(cv.polling_component_schema("60s"))
```